### PR TITLE
FILE_MODIFY_PERMISSIONS rector

### PR DESCRIPTION
**Pull request by Kbentham**
https://palantir.atlassian.net/browse/REC-68

### CODE REVIEW

File: deprecation-index.yml [hunk 1]
```diff
@@ -79,12 +79,18 @@
   Examples:
     - file_create_directory.php
     - FileCreateDirectoryNoUseStatement.php
-'FILE_EXISTS_REPLACE':
-  Rector: FileExistsReplaceRector.php
-  PHPStan: 'Call to deprecated constant FILE_EXISTS_REPLACE: Deprecated in drupal:8.7.0 and is removed from drupal:9.0.0. Use Drupal\Core\File\FileSystemInterface::FILE_EXISTS_REPLACE.'
```
Comment (damontgomery) [inline deprecation-index.yml]: @Kbentham, are we losing this record?

File: deprecation-index.yml [hunk 2]
```diff
@@ -79,12 +79,18 @@
...
+'FILE_MODIFY_PERMISSIONS':
+  Rector: FileModifyPermissionsRector.php
+  PHPStan: 'Call to deprecated constant FILE_MODIFY_PERMISSIONS: Deprecated in drupal:8.7.0 and is removed from drupal:9.0.0. Use Drupal\Core\File\FileSystemInterface::FILE_MODIFY_PERMISSIONS.'
   Examples:
-    - file_exists_replace.php
-    - FileExistsReplaceNoUseStatement.php
+    - file_modify_permissions.php
+    - FileModifyPermissionsNoUseStatement.php
+'FILE_CREATE_DIRECTORY':
```
Comment (damontgomery) [inline deprecation-index.yml]: Is this a duplicate? See line 76.

File: rector_examples_updated/file_modify_permissions.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+use Drupal\Core\File\FileSystemInterface;
```
Comment (damontgomery) [inline rector_examples_updated/file_modify_permissions.php:3]: @Kbentham, can you check if you are using rector 0.7.*? When I updated to that, I stopped getting `use` statements. We want them, but it wasn't working for me.

**Review by damontgomery — COMMENTED** [review]
Overall this looks good. Can you check the questions?

File: rector_examples_updated/file_modify_permissions.php (hunk 3 above)
Comment (Kbentham) [reply rector_examples_updated/file_modify_permissions.php:3]: I am on rector version v0.7.16

Comment (damontgomery) [reply rector_examples_updated/file_modify_permissions.php:3]: Thanks. The mysteries of the way this `use` statement is or isn't added continues to puzzle me.

**Pull request merged**